### ISSUE TITLE
Tweaking your code a bit

### ISSUE THREAD

**Issue by Raj2503** [stephengrice/synth-me#3]
Hello Stephen,

Hope you are well.
I am working on a project quite similar to yours, the only difference is that I have to make it for Hindi.

I have read quite a few research papers, and I understood the working of the System.

But I am having trouble with the implementation of the code.

It would be of great help if you guide me through the code or where, to begin with.

Regards 
Raj

**Comment by stephengrice** [comment]
Hi Raj! Thanks for your interest in this code!
I haven't worked on this project in a **long** time, but I can do my best to give you an intro to the codebase - hopefully it's helpful in some way.

* The entrypoint is in `synthme.py`. So, you run the entire program by typing `python3 synthme.py`. I made this project before I knew how to make a Python package with `setup.py` and all that, so it's pretty limited. I'd like to update it someday, but haven't had a chance.
* The main method calls `text_to_speech` in `tts.py`. This runs each major step of the program, which are:
  - Tokenize the message. This means that we take the raw string that the user types and split it into words. In this case, the tokenize function is pretty straightforward - it just splits based on spaces and punctuation using a regular expression.
  - The output of the previous step is passed to `words_to_phonemes`. This attempts to look up the pronunciation of the word in a Sqlite database (if use_pronunciation_dict is true). This allows you to hard-code the pronunciation of weird words in English that don't behave according to regular rules (which is a lot of words!). If it doesn't find any matches in Sqlite or use_pronunciation_dict is false, it then attempts to guess the pronuncation of the word. This "uneducated guess" is accomplished in the `phoneme_scan` method, where it looks for combinations of letters and pairs them with a phoneme (which is the smallest unit of pronunciation in a language). It starts with longer combinations of letters, like `oo` or `ou`, and falls back on single letters (and their default phonemes) if it can't find anything.
  - Now we have a list of phonemes, which is just logical sounds, like the things you see in the IPA pronunciation guide in a dictionary. The last step is very simple - we run `phonemes_to_sounds`, which goes through the list of phonemes, finds the `.wav` file associated with each one, and append it to our output sound file.

As a result of these steps, you should get a `.wav` file with some (terrible) text-to-speech audio! In my experience, it's pretty hard to understand what it's even saying, because the phoneme guesser is wrong more often than it's correct.

Does this help? Let me know if you have more questions!

**Comment by Raj2503** [comment]
Hey Stephen!
Your code was amazing.

We read and understood it through and it was amazing. We took inspiration from your code and did some changes to it and now it supports **Hindi** (Indic Language) as well.
If you would want to see it, here's the link - [Python-Hindi-TTS][1].

As you said Phoneme guesser was terrible at making a (un)educated guess. So we changed that as well. And it works fine now.
There are still some issues like the output contains a lot of **Areas of Silence** (as we call it in the TTS systems).
We are currently working on it.
And we are also thinking about expanding the Phoneme database as it contains only 44 files.

But seriously, big thanks to you. Without you, our Coursework would have not been possible.

Regards
Raj

[1]:https://github.com/Raj2503/Python-Hindi-TTS
